src/rules/no-anonymous-default-export.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * @fileoverview Rule to disallow anonymous default exports.
+ * @author Duncan Beevers
+ */
+
+const MAP = {
+  Literal: 'allow-literal',
+  ObjectExpression: 'allow-object-expression',
+  FunctionDeclaration: 'allow-anonymous-function-declaration',
+  ArrowFunctionExpression: 'allow-arrow-function-expression',
+  ClassDeclaration: 'allow-anonymous-class-declaration',
+}
+
+const FLAGS = []
+for (const nodeType in MAP) {
+  FLAGS.push(MAP[nodeType])
+}
+
+module.exports = {
+  meta: {
+    schema: {
+      type: 'array',
```

Comment:
Options are now handled as a single object. Details are spelled out in the rule description file.